plugins/chat/app/services/chat/restore_message.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Chat
+  # Service responsible for restoreing a trashed chat message
+  # for a channel and ensuring that the client and read state is
+  # updated.
+  #
+  # @example
+  #  Chat::RestoreMessage.call(message_id: 2, channel_id: 1, guardian: guardian)
+  #
+  class RestoreMessage
+    include Service::Base
+
+    # @!method call(message_id:, channel_id:, guardian:)
+    #   @param [Integer] message_id
+    #   @param [Integer] channel_id
+    #   @param [Guardian] guardian
+    #   @return [Service::Base::Context]
+
+    contract
+    model :message
+    policy :invalid_access
+    transaction do
+      step :restore_message
+      step :update_thread_reply_cache
+    end
+    step :publish_events
```

Comment:
This is so clean 😍

Reply:
Yes every time I use a service I really love it